### ISSUE TITLE
Console command doesn't work on pterodactyle server pls help !

### ISSUE THREAD

**Issue by Simon2122** [citizenfx/fivem#1976]
Hello, I have an artifact problem, since artifact 6247, all commands in the console do not work. The later versions all work but are outdated. Pls help me to find the problem!
![image](https://user-images.githubusercontent.com/65517485/236068409-b0e2da49-e176-4e36-9c6d-70b69562a3b4.png)

**Comment by ahcenezdh** [comment]
I think it was made on purpose, you have to use TxAdmin

**Comment by blattersturm** [comment]
> I think it was made on purpose, you have to use TxAdmin

what? that makes no sense.

**Comment by ahcenezdh** [comment]
Oh, I wasn't aware of that. For me, it seemed like it was done intentionally. Well, the last Linux build to support the Pterodactyl console is build 6188.

**Comment by blattersturm** [comment]
> Well, according to them, you did it on purpose.

And according to me, the person who supposedly 'did it on purpose', this wasn't done on purpose.

It's likely that this 'pterodactyle server' thing handles stdin in some unorthodox way that broke with some dependency update, and it doesn't really show of much maturity on _their_ end that instead of being constructive they spread a narrative that we're 'petty as fuck' and 'trying to get people to use txadmin' when txAdmin has _zero_ relationship to what their project is doing, and one can even deploy txAdmin using their management tool fine hypothetically: there's quite literally no conflict there.

**Comment by blattersturm** [comment]
In this case, the difference between 6231 and 6247 is listed here: https://github.com/citizenfx/fivem/compare/v1.0.0.6231...v1.0.0.6247

A likely cause here is the following commit responsible for terminal input: https://github.com/citizenfx/fivem/commit/bf86882d9420c739f8b4e6e1fa6104f1697ba29b

**Comment by Simon2122** [comment]
Any update ?

**Comment by parkervcp** [comment]
Just to make sure this is posted where people can see it. I have talked to Tabarra about updating the egg to work better for you guys. I talked to them more after that screenshot was taken. I was busy at the time and I am sorry I made it seem like it was broken on purpose.

We run all the containers in tty mode as seen here - https://github.com/pterodactyl/wings/blob/develop/environment/docker/container.go#L182

**Comment by ahcenezdh** [comment]
Hey sorry for the bump but any news on that?

**Comment by tabarra** [comment]
We can close this thread as #wontfix.  
FXServer's Node.js doesn't pipe stdin to txAdmin repass it to the child fxserver, and even if it did it might not be a good idea.  
Furthermore, at this point I'm not sure if there is any reason why the pterodactyl panel terminal would be preferable to txAdmin's, since now we both use xterm.js, and txAdmin should even have a better performance due to the Canvas renderer.  
  
Although, I do wish txAdmin could at least get the "quit" command that pterodactyl sends when shutting down the server so we could handle it gracefully.

**Comment by blattersturm** [comment]
I don't think this was a txAdmin-specific issue, though - there also was _another_ case with `stdin` handling the way Pterodactyl handled TTY input.

**Comment by tabarra** [comment]
I'm fairly sure I tested when I talked to Parker in the DMs, and stdin works without txAdmin.  
If it didn't work without txAdmin, people would have complained about it for sure as there would be no other way to do stdin outside of the in-game f8 console.

**Comment by gottfriedleibniz** [comment]
Since a clear regression was reported I can reopen this. It would be nice to know if this issue still persists as-well.

The [replxx](https://github.com/AmokHuginnsson/replxx) repository has not seen much maintenance in a few years, this repository references blattersturm's [fork](https://github.com/blattersturm/replxx/tree/master-cfx) to cherry-pick a few things, and other weird interactions have been [reported](https://github.com/citizenfx/fivem/issues/1827) elsewhere.

The game's afoot.

**Comment by parkervcp** [comment]
I haven't checked if stdin works without txadmin. I would have to install the server again to validate that.

**Comment by parkervcp** [comment]
I can confirm that the cli is broken with txadmin disabled.

**Comment by iWanheda** [comment]
I fixed this by adding `TERM=dumb` to the startup command.

<img width="377" height="168" alt="Image" src="https://github.com/user-attachments/assets/a95483a5-4c39-4175-a92a-9976bf981a28" />

What we get after:

<img width="1002" height="489" alt="Image" src="https://github.com/user-attachments/assets/abb0ba2f-74cb-47c1-b1a7-0eabb4b18e0d" />

Essentially, what it does is tell FXServer "you're running in a basic terminal that doesn't support fancy input features," so instead of trying to read your keystrokes one character at a time with raw terminal control (which breaks over Docker's attach stream), it just reads complete lines of text, type a command, hit enter, it processes it. Same way txAdmin sends commands to FXServer internally.

**Comment by parkervcp** [comment]
We can probably just add that as a default variable so it just works.
